File: 0258-add-digits/0258-add-digits.java
class Solution {
    public int loop(int num){
          int d=0;
          int s=0;
            while(num>0){
            d=num%10;
            s+=d;
            num/=10;
        }
        return s;

    }
    public int addDigits(int num) {
        int s1=0;
     while(num>9){
          s1=loop(num);
          num=s1;
     }
      
       
        return num;
    }
}

File: 0258-add-digits/0258-add-digits.c

int loop(int n){
    int rem=0,sum=0;
    while(n>0){
        rem=n%10;
        sum+=rem;
        n/=10;
    }
    return sum;
}
int addDigits(int num) {
if(num>=0 && num<=9)return num;
int s=0;
while(num>9){
s=loop(num);
num=s;
}
return num;

}